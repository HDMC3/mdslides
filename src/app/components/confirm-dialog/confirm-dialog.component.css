nb-card {
    max-width: 90vw;
}

.card-body {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
}

.alert-icon {
    width: 5rem;
    height: 5rem;
}

.card-footer {
    display: flex;
    justify-content: space-evenly;
}

.message {
    max-width: 40ch;
    margin: 0;
    text-align: center;
    padding: 1rem 0;
}

@media screen and (min-width: 768px) {
    nb-card {
        min-width: 350px;
    }
}