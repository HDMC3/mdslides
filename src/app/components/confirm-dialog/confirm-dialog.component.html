<nb-card>
    <nb-card-body>
        <div class="card-body">
            <nb-icon class="alert-icon" status="warning" icon="alert-triangle-outline"></nb-icon>
            <h6 class="message">{{ message }}</h6>
        </div>
    </nb-card-body>
    <nb-card-footer>
        <div class="card-footer">
            <button nbButton status="danger" (click)="close(false)">Cancelar</button>
            <button nbButton status="success" (click)="close(true)">{{ confirmButtonText }}</button>
        </div>
    </nb-card-footer>
</nb-card>