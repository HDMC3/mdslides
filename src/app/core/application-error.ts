export class ApplicationError extends Error {
    constructor(message: string, public title?: string) {
        super(message);
    }
}
