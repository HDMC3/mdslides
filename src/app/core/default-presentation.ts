import { v4 } from 'uuid';
import { Presentation } from '../data/interfaces/presentation';

export const getDefaultPresentation = (id: string): Presentation => {
    return {
        id,
        title: 'Titulo de presentacion',
        creation_date: new Date(),
        modification_date: new Date(),
        description: '',
        slides: [
            {
                id: v4(),
                name: 'Presentacion inicial',
                code: [
                    '---',
                    'style: |',
                    'section {',
                    '   display: flex;',
                    '   flex-direction: column;',
                    '   align-items: center;',
                    '   justify-content: center;',
                    '}',
                    '---',
                    '',
                    '# Inicio de la presentacion',
                    '# ðŸ˜„âœŒ'
                ]
            }
        ]
    };
};
