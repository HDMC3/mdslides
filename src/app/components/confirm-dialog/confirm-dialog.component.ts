import { Component, Input } from '@angular/core';
import { NbDialogRef } from '@nebular/theme';

@Component({
    selector: 'app-confirm-dialog',
    templateUrl: './confirm-dialog.component.html',
    styleUrls: ['./confirm-dialog.component.css']
})
export class ConfirmDialogComponent {

    @Input() message: string = 'Desea continuar?';
    @Input() confirmButtonText: string = 'Si';

    constructor(private nbDialogRef: NbDialogRef<any>) {
    }

    close(confirm: boolean) {
        this.nbDialogRef.close({ confirm });
    }

}
