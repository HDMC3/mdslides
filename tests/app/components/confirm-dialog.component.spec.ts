import { ComponentFixture, TestBed } from '@angular/core/testing';
import { NbEvaIconsModule } from '@nebular/eva-icons';
import { NbButtonModule, NbCardModule, NbDialogModule, NbDialogRef, NbIconModule, NbThemeModule } from '@nebular/theme';
import { Observable } from 'rxjs';

import { ConfirmDialogComponent } from '../../../src/app/components/confirm-dialog/confirm-dialog.component';

describe('ConfirmDialogComponent', () => {
    let component: ConfirmDialogComponent;
    let fixture: ComponentFixture<ConfirmDialogComponent>;
    let compiled: HTMLElement;

    const nbDialogRef = {
        close() { },
        componentRef: {},
        onBackdropClick: new Observable<MouseEvent>(),
        onClose: new Observable()
    };
    beforeEach(async() => {
        await TestBed.configureTestingModule({
            declarations: [ConfirmDialogComponent],
            imports: [
                NbThemeModule.forRoot({ name: 'default' }),
                NbCardModule,
                NbIconModule,
                NbEvaIconsModule,
                NbButtonModule,
                NbDialogModule.forRoot()
            ],
            providers: [{ provide: NbDialogRef, useValue: nbDialogRef }]
        })
            .compileComponents();
        // TestBed.inject<any>(NbDialogRef);
    });

    beforeEach(() => {
        fixture = TestBed.createComponent(ConfirmDialogComponent);
        component = fixture.componentInstance;
        fixture.detectChanges();
        compiled = fixture.nativeElement;
    });

    it('should create', () => {
        expect(component).toBeTruthy();
    });

    it('should match snapshot', () => {
        expect(compiled).toMatchSnapshot();
    });
});
