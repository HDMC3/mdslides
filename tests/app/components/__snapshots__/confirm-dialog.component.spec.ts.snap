// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmDialogComponent should match snapshot 1`] = `
<div
  id="root1"
>
  <nb-card>
    <nb-card-body>
      <div
        class="card-body"
      >
        <nb-icon
          class="alert-icon status-warning"
          icon="alert-triangle-outline"
          status="warning"
        >
          <svg
            class="eva eva-alert-triangle-outline"
            fill="currentColor"
            height="100%"
            viewBox="0 0 24 24"
            width="100%"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              data-name="Layer 2"
            >
              <g
                data-name="alert-triangle"
              >
                <rect
                  height="24"
                  opacity="0"
                  transform="rotate(90 12 12)"
                  width="24"
                />
                <path
                  d="M22.56 16.3L14.89 3.58a3.43 3.43 0 0 0-5.78 0L1.44 16.3a3 3 0 0 0-.05 3A3.37 3.37 0 0 0 4.33 21h15.34a3.37 3.37 0 0 0 2.94-1.66 3 3 0 0 0-.05-3.04zm-1.7 2.05a1.31 1.31 0 0 1-1.19.65H4.33a1.31 1.31 0 0 1-1.19-.65 1 1 0 0 1 0-1l7.68-12.73a1.48 1.48 0 0 1 2.36 0l7.67 12.72a1 1 0 0 1 .01 1.01z"
                />
                <circle
                  cx="12"
                  cy="16"
                  r="1"
                />
                <path
                  d="M12 8a1 1 0 0 0-1 1v4a1 1 0 0 0 2 0V9a1 1 0 0 0-1-1z"
                />
              </g>
            </g>
          </svg>
        </nb-icon>
        <h6
          class="message"
        >
          Desea continuar?
        </h6>
      </div>
    </nb-card-body>
    <nb-card-footer>
      <div
        class="card-footer"
      >
        <button
          aria-disabled="false"
          class="appearance-filled size-medium shape-rectangle status-danger"
          nbbutton=""
          status="danger"
          tabindex="0"
        >
          Cancelar
        </button>
        <button
          aria-disabled="false"
          class="status-success appearance-filled size-medium shape-rectangle"
          nbbutton=""
          status="success"
          tabindex="0"
        >
          Si
        </button>
      </div>
    </nb-card-footer>
  </nb-card>
</div>
`;
